lib/azure_jwt_auth/provider.rb

```diff
@@ -22,7 +29,11 @@ def initialize(uid, config_uri, validations={})
 
     def load_keys
       uri = URI(@config['jwks_uri'])
-      keys = JSON.parse(Net::HTTP.get(uri))['keys']
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      request = Net::HTTP::Get.new(uri.request_uri)
+      response = http.request(request)
+      keys = JSON.parse(response.body)['keys']
```

Comment:
This code is duplicated. I think that it's better to create a private function like this:

```
def make_get_request(url)
  uri = URI(url)
  http = Net::HTTP.new(uri.host, uri.port)
  http.use_ssl = true
  request = Net::HTTP::Get.new(uri.request_uri)
  response = http.request(request)
  JSON.parse(response.body)
end
```